src/main/java/io/lettuce/core/protocol/CommandHandler.java

```diff
@@ -840,6 +839,19 @@ private static long nanoTime() {
         return System.nanoTime();
     }
 
+    /**
+     * try discard read bytes when buffer usage reach {@code bufferUsageRatio / bufferUsageRatio + 1}
+     * @param buffer
+     */
+    private void discardReadBytesIfNecessary(ByteBuf buffer) {
+        int bufferUsageRatio = clientOptions.getBufferUsageRatio();
+        if ((float)buffer.writerIndex() / buffer.capacity() >= (float)bufferUsageRatio / (bufferUsageRatio + 1)) {
```

Comment:
We should rather check for `readerIndex()`. The written bytes don't express how much of the buffer was already consumed.